sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -533,7 +534,8 @@ class Dataset[T] private[sql](
    * Eagerly checkpoint a Dataset and return the new Dataset. Checkpointing can be used to truncate
    * the logical plan of this Dataset, which is especially useful in iterative algorithms where the
    * plan may grow exponentially. It will be saved to files inside the checkpoint
-   * directory set with `SparkContext#setCheckpointDir`.
+   * directory set with `SparkContext#setCheckpointDir`. 
+   * This is no-op in queries with streaming sources.
```

Comment:
:nit I found many comments already in Spark `This is a no-op if ...`, can you change yours to this fashion?